lib/findFirstMatch.js

```diff
@@ -17,5 +17,5 @@ export default function findFirstMatch(children, location) {
     }
   });
 
-  return match ? React.cloneElement(child, { location, computedMatch: match }) : null;
+  return match ? React.cloneElement(child, { location, computedMatch: match, key: child.key || location.key }) : null;
```

Comment:
whoops I didn't notice that here you're already making sure we don't get undefined keys